components/script/dom/validation.rs

```diff
@@ -1,5 +1,9 @@
 /* This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+use dom::validitystate::ValidationFlags;
 
-pub trait Validatable {}
+pub trait Validatable {
+    fn is_instance_validatable(&self) -> bool;
+    fn validate(&self, validate_flags: ValidationFlags) -> bool;
```

Comment:
Given that every implementation is a stub right now, I propose we do:
```rust
pub trait Validatable {
    fn is_instance_validatable(&self) -> bool { true }
    fn validate(&self, _validate_flags: ValidationFlags) -> bool { true }
}
```
and remove the stub implementations from the other files:
```rust
impl Validatable for HTMLTextAreaElement {}
```